calibration_guis/src/qnode.cpp

```diff
@@ -0,0 +1,127 @@
+/**
+ * @file /src/qnode.cpp
+ *
+ * @brief Ros communication central!
+ *
+ * @date February 2011
+ **/
+
+/*****************************************************************************
+** Includes
+*****************************************************************************/
+
+#include <ros/ros.h>
+#include <ros/network.h>
+#include <string>
+#include <std_msgs/String.h>
+#include <sstream>
+#include "../include/calibration_guis/qnode.hpp"
+
+/*****************************************************************************
+** Namespaces
+*****************************************************************************/
+
+namespace calibration_guis {
+
+/*****************************************************************************
+** Implementation
+*****************************************************************************/
+
+QNode::QNode(int argc, char** argv ) :
+	init_argc(argc),
+	init_argv(argv)
+	{}
+
+QNode::~QNode() {
+    if(ros::isStarted()) {
+      ros::shutdown(); // explicitly needed since we use ros::start();
+      ros::waitForShutdown();
+    }
+	wait();
+}
+
+bool QNode::init() {
+	ros::init(init_argc,init_argv,"calibration_guis");
+	if ( ! ros::master::check() ) {
+		return false;
+	}
+	ros::start(); // explicitly needed since our nodehandle is going out of scope.
+	ros::NodeHandle n;
+	// Add your ros communications here.
+	chatter_publisher = n.advertise<std_msgs::String>("chatter", 1000);
+	start();
+	return true;
+}
+
+bool QNode::init(const std::string &master_url, const std::string &host_url) {
+	std::map<std::string,std::string> remappings;
+	remappings["__master"] = master_url;
+	remappings["__hostname"] = host_url;
+	ros::init(remappings,"calibration_guis");
+	if ( ! ros::master::check() ) {
+		return false;
+	}
+	ros::start(); // explicitly needed since our nodehandle is going out of scope.
+	ros::NodeHandle n;
+	// Add your ros communications here.
+	chatter_publisher = n.advertise<std_msgs::String>("chatter", 1000);
+	start();
+	return true;
+}
+
+void QNode::run() {
+	ros::Rate loop_rate(1);
+	int count = 0;
+	while ( ros::ok() ) {
+
+		std_msgs::String msg;
+		std::stringstream ss;
+		ss << "hello world " << count;
```

Comment:
still need the hello world? Clean up and remove any unnecessary code before your pull requests